tests/ssl-opt.sh

```diff
@@ -14263,6 +14285,29 @@ run_test    "Handshake defragmentation on server: len=4, TLS 1.2" \
             -s "handshake fragment: 0 \\.\\. 4 of [0-9]\\+ msglen 4" \
             -s "waiting for more fragments (4"
 
+requires_openssl_3_x
+requires_protocol_version tls13
+requires_certificate_authentication
+run_test    "Handshake defragmentation on server: len=3, TLS 1.3" \
+            "$P_SRV debug_level=4 auth_mode=required" \
+            "$O_NEXT_CLI -tls1_3 -split_send_frag 3 -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
+            1 \
+            -s "<= parse client hello" \
+            -s "handshake message too short: 3" \
+            -s "SSL - An invalid SSL record was received"
+
+requires_openssl_3_x
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
+requires_certificate_authentication
+run_test    "Handshake defragmentation on server: len=3, TLS 1.3" \
+            "$P_SRV debug_level=4 auth_mode=required" \
+            "$O_NEXT_CLI -tls1_2 -split_send_frag 3 -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
+            1 \
+            -s "<= parse client hello" \
+            -s "handshake message too short: 3" \
+            -s "SSL - An invalid SSL record was received"
+
```

Comment:
Please also add a negative test where the server has `force_version=tls12` with a reasonable fragment size (anywhere between 4 and 40). We expect this to fail too, but please check what the message is.

Also, please document this in the ChangeLog entry: "Server-side, defragmentation of the ClientHello message is only supported if the server accepts TLS 1.3 (regardless of whether the ClientHello is 1.3 or 1.2). That is, servers configured (either at compile time or at runtime) to only accept TLS 1.2 will still fail the handshake if the ClientHello message is fragmented."